data/types/device/alarm/alarm.go

```diff
@@ -85,6 +120,15 @@ func (a *Alarm) Validate(validator structure.Validator) {
 		}
 		validator.String("statusId", a.StatusID).Using(data.IDValidator)
 	}
+	validator.String("ackStatus", a.AckStatus).OneOf(AckStatuses()...)
+	if a.EventID != nil {
+		validator.String("alarmLevel", a.AlarmLevel).Exists().OneOf(AlarmLevels()...)
+		validator.String("alarmCode", a.AlarmCode).Exists()
+		validator.String("alarmLabel", a.AlarmLabel).Exists()
+		validator.String("ackStatus", a.AckStatus).Exists()
+		timeValidator := validator.String("updateTime", a.UpdateTime)
+		timeValidator.AsTime(types.TimeFormat).Exists()
+	}
```

Comment:
From the spec :
> fields are mandatory if `alarmTYpe` is set to `handset`. They remain optional for any other values.

The logic here is that fields are validated when eventId exists. And updateTime (for thr time format) and alarmLevel(for the list of accepted values) are validated only if enventId is set.

Reply:
reworked based on alarmType.